core/src/main/java/io/grpc/Status.java

```diff
@@ -314,6 +317,32 @@ public static Status fromCodeValue(int codeValue) {
     }
   }
 
+  @SuppressWarnings("fallthrough")
+  private static Status fromCodeValue(byte[] asciiCodeValue) {
```

Comment:
optional: it might make sense to special case length 1, value 0. I would assume that's the status of 99,99...% of RPC calls. Could try if it helps.